corehq/apps/app_manager/tests/data/suite/child-module-entry-datums-added-usercase.xml

```diff
@@ -0,0 +1,30 @@
+<partial>
+  <entry>
+    <command id="m0-f0">
+      <text>
+        <locale id="forms.m0f0"/>
+      </text>
+    </command>
+    <instance id="casedb" src="jr://instance/casedb"/>
+    <instance id="commcaresession" src="jr://instance/session"/>
+    <session>
+      <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='gold-fish'][@status='open']" value="./@case_id" detail-select="m0_case_short" detail-confirm="m0_case_long"/>
+      <datum id="case_id_new_guppy_0" function="uuid()" />
+      <datum id="usercase_id" function="instance('casedb')/casedb/case[@case_type='commcare-user'][hq_user_id=instance('commcaresession')/session/context/userid][1]/@case_id"/>
+    </session>
+  </entry>
+  <entry>
+    <command id="m1-f0">
+      <text>
+        <locale id="forms.m1f0"/>
+      </text>
+    </command>
+    <instance id="casedb" src="jr://instance/casedb"/>
+    <instance id="commcaresession" src="jr://instance/session"/>
+    <session>
+      <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='gold-fish'][@status='open']" value="./@case_id" detail-select="m0_case_short"/>
+      <datum id="case_id_new_guppy_0" function="uuid()" />
+      <datum id="case_id_guppy" nodeset="instance('casedb')/casedb/case[@case_type='guppy'][@status='open'][index/parent=instance('commcaresession')/session/data/case_id]" value="./@case_id" detail-select="m1_case_short"  detail-confirm="m1_case_long"/>
```

Comment:
It might not be needed by the form but it is needed for navigation.

"""
CommCare prioritizes the most relevant piece of information to be determined by the user at any given time. For example: If a user selects that they want to enter a menu which contains three entry events which all require a pregnant mother's ID, CommCare will have the user select the pregnant mother and place that information into the Session before requesting that the user select a form.
"""" https://github.com/dimagi/commcare/wiki/Suite20#the-session